service-jehlomat/src/main/kotlin/services/DatabaseService.kt

```diff
@@ -128,6 +129,48 @@ class DatabaseService(
             .map(mapSyringeRow)
     }
 
+    fun selectSyringes(filter: SyringeFilter, organizationId: Int?): List<CSVExportSchema> {
+        val filterDsl = SyringeFilterTransformer.filterToDsl(filter, syringeCreatedByAlias, organizationId)
+
+        val selectColumns: MutableList<ColumnDeclaring<*>> = mutableListOf()
+        selectColumns.addAll(SyringeTable.columns)
+        selectColumns.addAll(LocationTable.columns)
+        selectColumns.addAll(syringeCreatedByAlias.columns)
+        selectColumns.addAll(syringeDemolishedByAlias.columns)
+        selectColumns.addAll(OrganizationTable.columns)
+        selectColumns.addAll(teamTableAlias.columns)
+
+        return databaseInstance.from(SyringeTable)
+            .innerJoin(LocationTable, LocationTable.id eq SyringeTable.locationId)
+            .leftJoin(syringeCreatedByAlias, syringeCreatedByAlias.userId eq SyringeTable.createdBy)
+            .leftJoin(syringeDemolishedByAlias, syringeDemolishedByAlias.userId eq SyringeTable.demolishedBy)
+            .leftJoin(OrganizationTable, syringeCreatedByAlias.organizationId eq OrganizationTable.organizationId)
+            .leftJoin(teamTableAlias, teamTableAlias.teamId eq syringeCreatedByAlias.teamId)
+            .select(selectColumns)
+            .where { filterDsl }
+            .orderBy(SyringeTable.id.asc())
+            .map { row ->
+                CSVExportSchema(
+                    id = row[SyringeTable.id]!!,
+                    createdTimestamp = row[SyringeTable.createdAt]!!,
+                    createdByEmail = row[syringeCreatedByAlias.email]!!,
+                    createdByUsername = row[syringeCreatedByAlias.username]!!,
+                    destroyedByEmail = row[syringeDemolishedByAlias.email],
+                    destroyedByUsername = row[syringeDemolishedByAlias.username],
+                    destroyedTimestamp = row[SyringeTable.demolishedAt],
+                    demolishingType = Demolisher.valueOf(row[SyringeTable.demolisherType]!!).czechName(),
+                    count = row[SyringeTable.count],
+                    gpsCoordinates = row[SyringeTable.gpsCoordinates],
+                    okres = getLocationName(row[LocationTable.okres]!!, "sph_okres"),
```

Comment:
The three getLocationName calls would be executed as three separate db queries for every row from the selected row set. That would take quite a long time, especially for the export of larger timespan.

Wouldn't be possible to implement it as a three left joins inside the main query?

Reply:
c9ac85de636fc91d199464d219f9de030c7cdf37